cincinnati/src/plugins/internal/graph_builder/openshift_secondary_metadata_parser/plugin.rs

```diff
@@ -343,14 +344,30 @@ impl OpenshiftSecondaryMetadataParserPlugin {
             .context(format!("Reading {:?}", &path))?;
         let string_version = String::from_utf8_lossy(&version);
 
-        if SUPPORTED_VERSIONS.contains(&string_version.trim()) {
-            Ok(string_version.into_owned())
-        } else {
-            Err(format_err!(
-                "unrecognized graph-data version {}; supported versions: {:?}",
+        if semver::Version::parse(&string_version).is_err() {
+            return Err(format_err!("{} is not a valid semver", string_version));
+        }
+
+        let req_version = semver::Version::parse(&string_version).unwrap();
+        let max_version = semver::Version::parse(MAX_SUPPORTED_VERSION).unwrap();
+
+        if req_version.major > max_version.major || req_version.major < max_version.major {
+            return Err(format_err!(
+                "unrecognized graph-data version {}; {:?} and compatible versions are supported",
                 string_version,
-                SUPPORTED_VERSIONS
-            ))
+                MAX_SUPPORTED_VERSION
+            ));
+        }
+
+        if req_version.minor > max_version.minor {
+            info!("graph-data version {} may have features which our {:?} implementation does not support", string_version, MAX_SUPPORTED_VERSION);
+            return Ok(MAX_SUPPORTED_VERSION.to_string());
+        } else {
+            info!(
+                "graph-data version {} is compatible with our {:?} implementation",
+                string_version, MAX_SUPPORTED_VERSION
+            );
+            return Ok(string_version.to_string());
```

Comment:
Looks like our happy-case used to be `Ok(string_version.into_owned())`, and now it's `Ok(string_version.to_string())`?  I'm not sure what the trade-offs are there, or if the two methods are basically synonyms.